Assets/Code/Scripts/Level generation scripts/TileGenerationScript.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class TileGenerationScript : MonoBehaviour
+{
+    public int Size { get; set; }
+    
+    public List<GameObject> tiles;
+
+    public float Width { get; set; }
+
+    public int Blocks { get; set; }
+
+    public abstract GameObject Generate(int i, int j);
```

Comment:
Serializable would be preferable

Reply:
I am not sure if serialized would work here since this class needs to get it's data from the parent controller.